README.rst

```diff
@@ -24,8 +24,12 @@ Install the bind package and start the bind service.
 ---------------
 
 Manage the bind configuration file.
-This state can generate some basic zone files if a `records` entry is found in the
-`available_zones`' declaration for the zone (see `pillar.example` for how to write these)
+
+`configured_zones` are declared in the `named` configuration.
+
+The `available_zones` key allows to manage the zone file, and possibly its content,
+if `soa` and `records` entries are found in the declaration for the zone. See
+`pillar.example` for how to write these.
```

Comment:
The options to manage zones are getting complex, so I'd try to explain them in a little more detail. Perhaps somethiing like:

----

## Management of zone files:
### `named.conf.local`

<zone> entries in `named.conf.local` will point to the file declared in

* `bind:configured_zones:<zone>:file` (this takes precedence)
* `bind:available_zones:<zone>:file`


### zone files

The `config.sls` state will  iterate on `bind:available_zones` and manage <zone> files for each <zone> that has bind:available_zones:<zone>:file` declared.

* If `bind:available_zones:<zone>:records` exist, a zone file will be created using those records (see pillar.example for more details)
* If `bind:available_zones:<zone>:records` is **NOT** declared, `bind:available_zones:<zone>:file` should point to an existing zone file that will be **sourced** by the formula.

### External zone files

To use an external tool to manage the <zone> file, simply declare the location of the zone file in  `bind:configured_zones:<zone>:file` and **don't** add any entry for the <zone> in  `bind:available_zones`

----

What do you think? Is it clearer?

Reply:
Also, to fully support this logic above, it seems some work is needed in the `named.conf.local` files for [redhat](https://github.com/saltstack-formulas/bind-formula/blob/master/bind/files/redhat/named.conf.local#L70-L74), [arch](https://github.com/saltstack-formulas/bind-formula/blob/master/bind/files/arch/named.conf.local#L52-L56), [freebsd](https://github.com/saltstack-formulas/bind-formula/blob/master/bind/files/arch/named.conf.local#L52-L56) and [suse](https://github.com/saltstack-formulas/bind-formula/blob/master/bind/files/suse/named.conf.local#L65-L69), as the only one following this logic seems to be [debian's](https://github.com/saltstack-formulas/bind-formula/blob/master/bind/files/debian/named.conf.local#L76-L81). 

If you agree and find these comments rational, would you add those changes?

Reply:
>Regarding your last message, I indeed see some differences in the Debian file but I'm not sure about how it has to be handled.

I think that following/reproducing Debian's pattern (with the conditional for the existence of the `bind:configured_zones:<zone>:file`) it will behave as the proposed README mentions, consistently across distros?

Reply:
@javierbertoli Sorry but would you suggest a diff for this? Because I don't really get it. :-/

Reply:
I think that this diff would do what I mention (the same logic inside views should be applied, but that'd probably be worth another issue/PR)

```diff
diff --git a/bind/files/arch/named.conf.local b/bind/files/arch/named.conf.local
index a7fb87c..b372c11 100644
--- a/bind/files/arch/named.conf.local
+++ b/bind/files/arch/named.conf.local
@@ -50,8 +50,16 @@ zone "{{ key }}" {
 {%- endmacro %}
 
 {% for key, args in salt['pillar.get']('bind:configured_zones', {}).items() -%}
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":file") -%}
+{%- set file = salt['pillar.get']("bind:configured_zones:" + key + ":file") %}
+{% else %}
 {%- set file = salt['pillar.get']("bind:available_zones:" + key + ":file") %}
+{%- endif -%}
+
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":masters") -%}
+{%- set masters = salt['pillar.get']("bind:configured_zones:" + key + ":masters") %}
+{% else %}
 {%- set masters = salt['pillar.get']("bind:available_zones:" + key + ":masters") %}
+{%- endif -%}
 {{ zone(key, args, file, masters) }}
 {% endfor %}
-
diff --git a/bind/files/freebsd/named.conf.local b/bind/files/freebsd/named.conf.local
index 43e9fda..3464a1c 100644
--- a/bind/files/freebsd/named.conf.local
+++ b/bind/files/freebsd/named.conf.local
@@ -68,8 +68,17 @@ include "{{ map.default_zones_config }}";
 {%- endif %}
 
 {% for key, args in salt['pillar.get']('bind:configured_zones', {}).items() -%}
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":file") -%}
+{%- set file = salt['pillar.get']("bind:configured_zones:" + key + ":file") %}
+{% else %}
 {%- set file = salt['pillar.get']("bind:available_zones:" + key + ":file") %}
+{%- endif -%}
+
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":masters") -%}
+{%- set masters = salt['pillar.get']("bind:configured_zones:" + key + ":masters") %}
+{% else %}
 {%- set masters = salt['pillar.get']("bind:available_zones:" + key + ":masters") %}
+{%- endif -%}
 {{ zone(key, args, file, masters) }}
 {% endfor %}
 
diff --git a/bind/files/redhat/named.conf.local b/bind/files/redhat/named.conf.local
index fc5907a..eb59deb 100644
--- a/bind/files/redhat/named.conf.local
+++ b/bind/files/redhat/named.conf.local
@@ -68,8 +68,17 @@ zone "{{ key }}" IN {
 {%- endmacro %}
 
 {% for key, args in salt['pillar.get']('bind:configured_zones', {}).items() -%}
-{%- set file = args.get('file', salt['pillar.get']("bind:available_zones:" + key + ":file")) %}
-{%- set masters = args.get('masters', salt['pillar.get']("bind:available_zones:" + key + ":masters")) %}
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":file") -%}
+{%- set file = salt['pillar.get']("bind:configured_zones:" + key + ":file") %}
+{% else %}
+{%- set file = salt['pillar.get']("bind:available_zones:" + key + ":file") %}
+{%- endif -%}
+
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":masters") -%}
+{%- set masters = salt['pillar.get']("bind:configured_zones:" + key + ":masters") %}
+{% else %}
+{%- set masters = salt['pillar.get']("bind:available_zones:" + key + ":masters") %}
+{%- endif -%}
 {{ zone(key, args, file, masters) }}
 {% endfor %}
 
diff --git a/bind/files/suse/named.conf.local b/bind/files/suse/named.conf.local
index e0a2d4d..5f377e3 100644
--- a/bind/files/suse/named.conf.local
+++ b/bind/files/suse/named.conf.local
@@ -63,8 +63,17 @@ zone "{{ key }}" IN {
 {%- endmacro %}
 
 {% for key, args in salt['pillar.get']('bind:configured_zones', {}).items() -%}
-{%- set file = args.get('file', salt['pillar.get']("bind:available_zones:" + key + ":file")) %}
-{%- set masters = args.get('masters', salt['pillar.get']("bind:available_zones:" + key + ":masters")) %}
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":file") -%}
+{%- set file = salt['pillar.get']("bind:configured_zones:" + key + ":file") %}
+{% else %}
+{%- set file = salt['pillar.get']("bind:available_zones:" + key + ":file") %}
+{%- endif -%}
+
+{%- if salt['pillar.get']("bind:configured_zones:" + key + ":masters") -%}
+{%- set masters = salt['pillar.get']("bind:configured_zones:" + key + ":masters") %}
+{% else %}
+{%- set masters = salt['pillar.get']("bind:available_zones:" + key + ":masters") %}
+{%- endif -%}
 {{ zone(key, args, file, masters) }}
 {% endfor %}
```

wdyt?

Reply:
Thanks @javierbertoli . But indeed, I would see this in another PR if it's ok for you, can I let you open another one?